utilities/api/python_client_semgrep_api.py

```diff
@@ -63,16 +64,20 @@ def get_all_findings(projects, headers):
     """
     for project in projects['projects']:
         project_name = project['name']
+        primary_branch = project['default_branch']
         print("Getting findings for: " + project_name)
-        get_findings_per_project(deployment_slug, project_name, headers)
-    
+        get_findings_per_project(deployment_slug, project_name, primary_branch, headers)    
 
-def get_findings_per_project(deployment_slug, project, headers):
+def get_findings_per_project(deployment_slug, project, primary_branch, headers):
     """
     Gets all findings for a project, and writes them to a file.
     The file format is equivalent to what the API would return if it weren't paginated.
     """
-    project_findings = retrieve_paginated_data(f"{BASE_URL}/{deployment_slug}/findings?repos={project}&dedup=false", "findings", 3000, headers=headers)
+    findings_url = f"{BASE_URL}/{deployment_slug}/findings?repos={project}&dedup=true"
+    if USE_PRIMARY_BRANCH_PARAM:
+        findings_url = f"{findings_url}&ref={primary_branch}"
```

Comment:
I think in this case maybe dedup can be false, as well?

Reply:
Yes!